stage1_fly/run/main.go

```diff
@@ -0,0 +1,386 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	stage1commontypes "github.com/coreos/rkt/stage1_common/types"
+
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/appc/spec/schema"
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/appc/spec/schema/types"
+
+	"github.com/coreos/rkt/common"
+	"github.com/coreos/rkt/pkg/sys"
+)
+
+const (
+	flavor = "fly"
+)
+
+type flyMount struct {
+	HostPath         string
+	TargetPrefixPath string
+	RelTargetPath    string
+	Fs               string
+	Flags            uintptr
+}
+
+type volumeMountTuple struct {
+	V types.Volume
+	M schema.Mount
+}
+
+var (
+	debug       bool
+	netList     common.NetList
+	interactive bool
+	mdsToken    string
+	localhostIP net.IP
+	localConfig string
+)
+
+func getHostMounts() (map[string]struct{}, error) {
+	hostMounts := map[string]struct{}{}
+
+	mi, err := os.Open("/proc/self/mountinfo")
+	if err != nil {
+		return nil, err
+	}
+	defer mi.Close()
+
+	sc := bufio.NewScanner(mi)
+	for sc.Scan() {
+		var (
+			discard    string
+			mountPoint string
+		)
+
+		_, err := fmt.Sscanf(sc.Text(),
+			"%s %s %s %s %s",
+			&discard, &discard, &discard, &discard, &mountPoint,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		hostMounts[mountPoint] = struct{}{}
+	}
+	if sc.Err() != nil {
+		return nil, fmt.Errorf("problem parsing mountinfo: %v", sc.Err())
+	}
+	return hostMounts, nil
+}
+
+func init() {
+	flag.BoolVar(&debug, "debug", false, "Run in debug mode")
+
+	// TODO: implement processing of the following flags or give not implemented warnings
+	flag.Var(&netList, "net", "Setup networking")
+	flag.BoolVar(&interactive, "interactive", true, "The pod is interactive")
+	flag.StringVar(&mdsToken, "mds-token", "", "MDS auth token")
+	flag.StringVar(&localConfig, "local-config", common.DefaultLocalConfigDir, "Local config path")
+
+	localhostIP = net.ParseIP("127.0.0.1")
+	if localhostIP == nil {
+		panic("localhost IP failed to parse")
+	}
+}
+
+func lookupPath(bin string, paths string) (string, error) {
+	pathsArr := filepath.SplitList(paths)
+	for _, path := range pathsArr {
+		binPath := filepath.Join(path, bin)
+		binAbsPath, err := filepath.Abs(binPath)
+		if err != nil {
+			return "", fmt.Errorf("unable to find absolute path for %s", binPath)
+		}
+		d, err := os.Stat(binAbsPath)
+		if err != nil {
+			continue
+		}
+		// Check the executable bit, inspired by os.exec.LookPath()
+		if m := d.Mode(); !m.IsDir() && m&0111 != 0 {
+			return binAbsPath, nil
+		}
+	}
+	return "", fmt.Errorf("unable to find %q in %q", bin, paths)
+}
+
+func withClearedCloExec(lfd int, f func() error) error {
```

Comment:
Can this be moved to `stage1_common/common.go`? And get both implementation of stage1 to use it instead?

Same with `writePpid()` below.